### PR TITLE
Update docs for multiple optimizers in 2.0

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

For a summary of changes to how multiple optimizers are handled in the new version, see #16539. 
This PR brings the necessary documentation changes.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source-pytorch/common/optimization.rst [hunk 1]
```diff
@@ -83,113 +63,6 @@ Gradient Accumulation
 .. include:: ../common/gradient_accumulation.rst
 
 
-Use Multiple Optimizers (like GANs)
```
Comment (awaelchli) [inline docs/source-pytorch/common/optimization.rst:86]: The same section already exists in manual_optimization.rst below

File: docs/source-pytorch/common/optimization.rst [hunk 2]
```diff
@@ -83,113 +63,6 @@ Gradient Accumulation
...
-                opt.zero_grad()
-                loss.backward()
-                opt.step()
-
-        for lr_scheduler in lr_schedulers:
-            lr_scheduler.step()
-
-
-Step Optimizers at Arbitrary Intervals
-=======================================
```
Comment (awaelchli) [inline docs/source-pytorch/common/optimization.rst:131]: See the new section in manual_optimization.rst

File: docs/source-pytorch/common/optimization.rst (hunk 1 above)
Comment (Borda) [reply docs/source-pytorch/common/optimization.rst:86]: can we just add cross-link?

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/model/manual_optimization.rst [hunk 3]
```diff
@@ -285,6 +282,32 @@ If you want to call schedulers that require a metric value after each epoch, con
...
+In manual optimization, you are free to ``step()`` one optimizer more often than another one.
+For example, here we step the optimizer for the *discriminator* weights twice as often as the optimizer for the *generator*.
+
+.. testcode:: python
+
+    # Alternating schedule for optimizer steps (e.g. GANs)
+    def training_step(self, batch, batch_idx):
+
+        # update discriminator every other step
+        d_opt.zero_grad()
```
Comment (carmocca) [inline docs/source-pytorch/model/manual_optimization.rst:298]: Missing the `self.optimizers()` call

**Review by carmocca — APPROVED** [review]

File: docs/source-pytorch/common/optimization.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source-pytorch/common/optimization.rst:86]: done

Comment (awaelchli) [reply docs/source-pytorch/common/optimization.rst:86]: unfortunately didn't work, because the page is included via .. include and then the label is duplicated. A limitation of include directive.
